core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -428,7 +429,19 @@ public void renameTable(SessionContext context, TableIdentifier from, TableIdent
     Endpoint.check(endpoints, Endpoint.V1_RENAME_TABLE);
     checkIdentifierIsValid(from);
     checkIdentifierIsValid(to);
+    try {
+      renameInternal(context, from, to);
+    } catch (NoSuchNamespaceException e) {
+      if (name().equals(to.namespace().level(0)) && to.namespace().length() > 1) {
+        Namespace namespace =
+            Namespace.of(Arrays.copyOfRange(to.namespace().levels(), 1, to.namespace().length()));
+        TableIdentifier toWithStrippingCatalog = TableIdentifier.of(namespace, to.name());
+        renameInternal(context, from, toWithStrippingCatalog);
+      }
```

Comment:
this needs to `throw e` in case we didn't reach the if block. Also I believe we need to apply the same logic to `renameView`